### DISCUSSION TITLE
How to assign physical thumb pistons to GrandOrgue

### DISCUSSION THREAD

**Discussion by macsean3** [GrandOrgue/grandorgue#1103] [Help] [answered]
Hello everyone,

Our chapel just got a new 2-manual Arnold organ from Minnesota that I have been using in tandem with GrandOrgue for the past week or so.

Is there a way to assign the physical thumbs pistons (there are 10, as well as Set and Cancel/Clear) on the organ to work with GrandOrgue?  The only way I can get any Presets to work is in the actual software in the Generals window.  I have to leave that window open if I need to change Presets.

I'm using the Friesach Organ from Piotr Grabowski, which is very nice.

Thanks in advance.

Regards,
Sean...

**Comment by macsean3** [comment, marked as answer]
I'm an idiot.  'Simply added them from the Generals by right-clicking on the numbered presets in the software.  Now if I could get the software to recognize the Crescendo Pedal...

**Comment by kerkovits** [comment]
> Now if I could get the software to recognize the Crescendo Pedal...

Does a right-click on the pedal solve it? It pops a window up, where you can listen to MIDI-events. You should move the physical pedal to the maximal and minimal positions, as the program asks.

**Comment by macsean3** [comment]
I’ve been doing that for weeks and nothing happens.  On a Mac, you get that little wristwatch with the spinning second hand.  That’s all I get, just the spinning second hand.  

It’s pretty frustrating.

**Comment by macsean3** [comment]
And I’ve tried both Listen for Event as well as Detect Complex Midi Setup and it just sits there, unable to connect.

**Reply by willeke1234** [reply]
Have you tried "Log MIDI Events" (Audio/Midi menu)?

**Reply by macsean3** [reply]
I haven’t.  Does that just give me data that indicates that a MIDI signal is coming from the pedal?

**Reply by kerkovits** [reply]
Yes, it does. If no data is logged, then the error is probably in your crescendo pedal or the MIDI interface. If data is logged, then your pedal probably emits non-standard MIDI messages. In the latter case, you might be able to manually type the message into the pop-up window.

**Reply by macsean3** [reply]
I did Log MIDI Events this morning.  Where would I find the log data on the drive?  Does GO automatically designate a folder where the data resides?

**Reply by kerkovits** [reply]
No. It should pop up in a new window as soon as the MIDI event happens. If you press any key, it should appear. If you move the pedal, you may also encounter new messages in that window.

**Reply by macsean3** [reply]
I moved the pedal and nothing happened.  I guess I could request a standard MIDI message replacement pedal?  I don’t understand why it wouldn’t be a standard MIDI pedal from every manufacturer.

**Reply by kerkovits** [reply]
I my opinion, your crescendo pedal does not seem to be 'non-standard' considering the symptoms. I think that it is not working at all. Did you check all the wires and all connectors? Is not it disabled by some firmware settings of the console?

**Reply by macsean3** [reply]
I’ll check the connections again, though the electronics guy who was at the church set it all up and said everything was hooked up correctly.  He even called the company during the installation, asking if there were a trick to getting the pedal to work.  How would I check firmware settings?

**Reply by macsean3** [reply]
I spoke with the manufacturer on the phone this morning and he thought it might be the skinny ribbon cable that attaches to the pedal. At any rate, he's sending me an entirely new pedal in the morning with replacement cabling.  I hope that solves the issue.

**Comment by macsean3** [comment]
I thought perhaps if I installed the latest upgrade of the GrandOrgue software that things might change (v. 3.6.4-1 for OSX).  When I opened it, everything was in French.  So I’ll add that to the list of questions.  🤔

**Comment by macsean3** [comment]
I shall try your suggestion.  Thanks so much for your insights, I really appreciate them.
